java/ql/lib/semmle/code/java/security/PathSanitizer.qll

```diff
@@ -352,3 +352,43 @@ private class FileGetNameSanitizer extends PathInjectionSanitizer {
     )
   }
 }
+
+/** Holds if `expr` may be null. */
+private predicate maybeNull(Expr expr) {
+  exists(DataFlow::Node src, DataFlow::Node sink |
+    src.asExpr() = nullExpr() and
+    sink.asExpr() = expr
+  |
+    DataFlow::localFlow(src, sink)
+  )
+}
+
+/** Holds if `g` is a guard that checks for `..` components. */
+private predicate pathTraversalGuard(Guard g, Expr e, boolean branch) {
+  branch = g.(PathTraversalGuard).getBranch() and
+  localTaintFlowToPathGuard(e, g)
```

Comment:
I think the comment above the original lines should be moved here, because it is there to explain why local taint flow is used.

Reply:
Fixed in https://github.com/github/codeql/pull/18504/commits/4a4585a526cd21bb2f2b936b91861f8844f4734c. Thanks for catching that.